### DISCUSSION TITLE
Recent Update: Discord introduced a Rate-Limit on Slash Commands

### DISCUSSION THREAD

**Discussion by samjesus8** [DSharpPlus/DSharpPlus#1565] [Q&A] [answered]
# Rate Limit on Slash Commands being registered

![image](https://github.com/DSharpPlus/DSharpPlus/assets/98812930/4c0e842a-f368-46aa-b399-91096aede81b)

Just recently people have been coming to me with issues of Ratelimit on their slash commands being registered and we know that Discord put a rate-limit of 200 on this. So a simple question, is there any workaround to this because this prevents bots from even starting up??

# Error Message in the console

![image](https://github.com/DSharpPlus/DSharpPlus/assets/98812930/93050c52-4087-4f3c-a1ef-42fc6672e902)

## Possible solutions I can think of

- What I have thought of is hooking into the CommandErrored event and just displaying a message that the limit has been reached, like a cooldown almost
- I doubt you can do this, but before the bot starts up you can check if the rate limit is active, and based on that, register the slash commands or don't.

**Comment by akiraveliara** [comment, marked as answer]
this seems to be caused by https://github.com/discord/discord-api-docs/issues/6188, in which case there's nothing we can really do

**Reply by samjesus8** [reply]
I thought so, well thanks for the reply anyway, I thought It would be best to come to you guys before trying to think of any actual solutions
